libmachine/libmachine.go

```diff
@@ -54,11 +56,22 @@ func NewClient(storePath, certsDir string) *Client {
 }
 
 func (api *Client) NewHost(driverName string, rawDriver []byte) (*host.Host, error) {
+	// we can hide the container runtime parameter here
+	// until we can refactor the machine API or all the drivers structs
+	badHack := struct {
```

Comment:
It is terrible.. but just an idea:
We could pass the reference for the container runtime inside the rawDriver at the moment, instead of passing it as an argument as I tried here: https://github.com/kubernetes/minikube/pull/16459/files under pkg/libmachine/libmachine/libmachine.go:NewHost(), or instead of calling viper and getting it from the machine's profile name in the minikube dir, as it is done right now.

however this impacts each driver's code. so the driver part should be taken care of before the provisioner one perhaps?

I don't really know how this should look like:
KIC driver has a reference to the container runtime inside the driver's struct.
I think other drivers don't care about container runtime.